### PR TITLE
Fix: Dynamic Canvas Height based on bottom most widget.

### LINKED ISSUE #2962: [Bug] The default height of the canvas is larger than the screen area
## Description

A user has a single table on the screen but the canvas has a large scroll area below it

### PR DESCRIPTION
**Pull request by marks0351**
## Description

Calculate and set the initial height of pages dynamically.
Fixes #2962 

## Type of change

> Please delete options that are not relevant.

- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

> Please describe the tests that you ran to verify your changes. Provide instructions, so we can reproduce. 
> Please also list any relevant details for your test configuration.

- Test A
- Test B

## Checklist:

- [ ] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: app/client/src/pages/Editor/Explorer/Pages/PageGroup.tsx [hunk 1]
```diff
@@ -47,7 +48,10 @@ export const ExplorerPageGroup = memo((props: ExplorerPageGroupProps) => {
       "Page",
       pages.map((page: Page) => page.pageName),
     );
-    dispatch(createPage(params.applicationId, name));
+    const defaultPageLayouts = [
+      { dsl: extractCurrentDSL(), layoutOnLoadActions: [] },
```
Comment (riodeuno) [inline app/client/src/pages/Editor/Explorer/Pages/PageGroup.tsx:52]: Could you add a comment here explaining the reasoning for this? I believe the new DSL comes from servers. Is there something new that the client does before it creates a new page?

File: app/client/src/utils/WidgetPropsUtils.tsx [hunk 2]
```diff
@@ -337,13 +339,41 @@ function migrateOldChartData(currentDSL: ContainerWidgetProps<WidgetProps>) {
...
+    [widgetId: string]: FlattenedWidgetProps;
+  } = {},
+  presentMinimumHeight = CANVAS_DEFAULT_HEIGHT_PX,
+) => {
+  let minmumHeight = presentMinimumHeight;
+  const nextAvailableRow = nextAvailableRowInContainer(
+    MAIN_CONTAINER_WIDGET_ID,
+    canvasWidgets,
+  );
+  const screenHeight = window.innerHeight;
```
Comment (riodeuno) [inline app/client/src/utils/WidgetPropsUtils.tsx:353]: Won't this include the header and the padding below the header in the height? If so, will the buffer defined here, be enough?

File: app/client/src/pages/Editor/Explorer/Pages/PageGroup.tsx (hunk 1 above)
Comment (marks0351) [reply app/client/src/pages/Editor/Explorer/Pages/PageGroup.tsx:52]: Will add the comment. we did a recent change to api to accept layout from client. This way we wud pre calculate the minheight of the canavs.

File: app/client/src/utils/WidgetPropsUtils.tsx (hunk 2 above)
Comment (marks0351) [reply app/client/src/utils/WidgetPropsUtils.tsx:353]: Header + Canvas layout control height in Edit mode is almost equal(differ by 2 or 3px) to Header + Pages tab in View mode.
I gave the buffer just to not block out any last row rendered elements.

**Review by riodeuno — APPROVED** [review]

**Pull request merged**